src/main/java/org/openbase/bco/bcozy/controller/powerterminal/PowerChartVisualizationController.java

```diff
@@ -0,0 +1,297 @@
+            }, 1, refreshTimeout, TimeUnit.MINUTES);
+        } catch (NotAvailableException ex) {
+            ExceptionPrinter.printHistory("Could not refresh power chart data", ex, LOGGER);
+        }
+    }
+
+
+    /**
+     * Initializes the previous energy consumption
+     *
+     * @return List of ChartData with previous Energy Consumption
+     */
+    private List<ChartData> initializePreviousEntries(String interval, long startTime, long endTime) {
+        List<ChartData> datas = new ArrayList<ChartData>();
+        int change = 0;
+        olddataTime = new Timestamp(TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()));
+        try {
+            List<FluxTable> energy = InfluxDBHandler.getAveragePowerConsumptionTables(
+                    interval, TimeUnit.MILLISECONDS.toSeconds(startTime), TimeUnit.MILLISECONDS.toSeconds(endTime), "consumption");
+            for (FluxTable fluxTable : energy) {
+                List<FluxRecord> records = fluxTable.getRecords();
+                for (FluxRecord fluxRecord : records) {
+                    if (fluxRecord.getValueByKey("_value") == null) {
+                        ChartData temp = new ChartData(String.valueOf(change), 0, Tile.ORANGE);
+                        temp.setName(String.valueOf(change));
+                        datas.add(temp);
+                    } else {
+                        ChartData temp = new ChartData(String.valueOf(change), (double) fluxRecord.getValueByKey("_value") / dataStep, Tile.ORANGE);
+                        temp.setName(String.valueOf(change));
+                        datas.add(temp);
+                    }
+
+                }
+                change++;
+            }
+        } catch (CouldNotPerformException e) {
+            e.printStackTrace();
+        }
+        return datas;
+    }
+
+    /**
+     * Connects the given chart attribute properties to the chart by creating listeners incorporating the changes
+     * into the chart
+     * @param chartStateModel StateModel that describes the state of the chart as configured by other panes
+     */
+    public void initChartState(ChartStateModel chartStateModel) {
+
+        this.chartStateModel = chartStateModel;
+
+        chartStateModel.visualizationTypeProperty().addListener(
+                (ChangeListener<? super VisualizationType>) (dont, care, newVisualizationType) -> {
+                    this.visualizationType = newVisualizationType;
+                    setChartType(newVisualizationType);
+                });
+
+        chartStateModel.dateRangeProperty().addListener((ChangeListener<? super DateRange>) (dont, care, newDateRange) -> {
+            System.out.println("===========> Date Picked!");
+            setChartType(this.visualizationType);
+        });
+
+    }
+
+    private void setChartType(VisualizationType newVisualizationType) {
+        pane.getChildren().clear();
+        Node node;
+        if (newVisualizationType == VisualizationType.WEBVIEW) {
```

Comment:
use a switch case 
* WEBVIEW...
* default...

Reply:
Resolved in 05c6bb3